integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/ingest/CreateElasticsearchBackedHiveTable.java

```diff
@@ -0,0 +1,285 @@
+        validateStringProperty("category name", categoryName, session, flowFile);
+
+        final String nodes = context.getProperty(NODES).evaluateAttributeExpressions(flowFile).getValue();
+        validateStringProperty("elasticsearch nodes", nodes, session, flowFile);
+
+        final String indexString = context.getProperty(FIELD_INDEX_STRING).evaluateAttributeExpressions(flowFile).getValue();
+        validateStringProperty("index string", indexString, session, flowFile);
+
+        List<String> hiveStatements = getHQLStatements(columnSpecs, nodes, feedName, categoryName, useWan, autoIndex, idField, jarUrl, indexString);
+        final ThriftService thriftService = context.getProperty(THRIFT_SERVICE).asControllerService(ThriftService.class);
+
+        executeStatements(session, flowFile, hiveStatements.toArray(new String[hiveStatements.size()]), thriftService);
+
+    }
+
+    private void validateArrayProperty(String propertyName, Object[] values, ProcessSession session, FlowFile flowFile) {
+        if (values == null || values.length == 0) {
+            getLog().error("Missing " + propertyName + " specification");
+            session.transfer(flowFile, IngestProperties.REL_FAILURE);
+            return;
+        }
+    }
+
+    private void validateStringProperty(String propertyName, String propertyValue, ProcessSession session, FlowFile flowFile) {
+        if (propertyValue == null || propertyValue.isEmpty()) {
+            getLog().error("Missing " + propertyName + " parameter");
+            session.transfer(flowFile, IngestProperties.REL_FAILURE);
+            return;
+        }
+    }
+
+    public String generateHQL(String columnsSQL, String nodes, String feedName, String categoryName, String useWan, String autoIndex, String idField) {
+        StringBuilder sb = new StringBuilder();
+        sb.append("CREATE EXTERNAL TABLE IF NOT EXISTS ")
+            .append(categoryName)
+            .append(".")
+            .append(feedName)
+            .append("_index")
+            .append(" (")
+            .append(columnsSQL)
+            .append(") ")
+            .append("STORED BY 'org.elasticsearch.hadoop.hive.EsStorageHandler' TBLPROPERTIES('es.resource' = '")
+            .append(categoryName)
+            .append("/")
+            .append(feedName)
+            .append("', 'es.nodes' = '")
+            .append(nodes)
+            .append("', 'es.nodes.wan.only' = '")
+            .append(useWan)
+            .append("', 'es.index.auto.create' = '")
+            .append(autoIndex);
+
+        if (idField != null && !idField.isEmpty()) {
+            sb.append("', 'es.mapping.id' = '")
+                .append(idField);
+        }
+
+        sb.append("')");
+
+        return sb.toString();
+    }
+
+    public List<String> getHQLStatements(ColumnSpec[] columnSpecs, String nodes, String feedName, String categoryName, String useWan, String autoIndex, String idField,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Method has 9 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)